site/en/blog/immutable-document-domain/index.md

```diff
@@ -0,0 +1,197 @@
+`postMessage()` or Channel Messaging API to communicate cross-origin. If a
+website relies on same-origin policy relaxation via `document.domain` to
+function correctly, it will need to send an `Origin-Agent-Cluster: ?0` header
+along with all documents that require that behavior (note that `document.domain`
+has no effect if only one document sets it).
+
+## Why make `document.domain` immutable?
+
+Many websites set `document.domain` to allow communication between [same-site
+but cross-origin](https://web.dev/same-site-same-origin/) pages. 
+
+{% Aside 'key-term' %}
+
+Sites which have the same
+[eTLD+1](https://web.dev/same-site-same-origin/#:~:text=the%20whole%20site%20name%20is%20known%20as%20the%20etld%2B1)
+but are different subdomains are considered "same-site but cross-origin".
+
+{% endAside %}
+
+Here's how it's used:
+
+Let's say a page on `https://parent.example.com` embeds an iframe page from
+`https://video.example.com`. These pages have the same eTLD+1 (`example.com`)
+with different subdomains. By setting both pages' `document.domain` to
+`'example.com'`, the two origins can be treated as if they are the same-origin
+by the browser.
+
+On `https://parent.example.com`:
+
+```js
+console.log(document.domain); // "parent.example.com"
+document.domain = 'example.com';
+console.log(document.domain); // "example.com"
+```
+
+On `https://video.example.com`:
+
+```js
+console.log(document.domain); // "video.example.com"
+document.domain = 'example.com';
+console.log(document.domain); // "example.com"
+```
+
+On `https://parent.example.com`, do a cross origin DOM manipulation against
+`https://video.example.com`.
+
+Websites set `document.domain` to make it possible for same-site documents to
+communicate more easily. Because this change [relaxes the same-origin
+policy](https://html.spec.whatwg.org/multipage/origin.html#relaxing-the-same-origin-restriction),
+the parent page will be able to access the iframe's document and traverse the
+DOM tree, and vice versa.
+
+At a glance, this is a convenient technique, but this actually introduces a
+security risk.
+
+### Security concerns with `document.domain`
+
+Security concerns have led to the spec for `document.domain` to [warn users to
+avoid
+usage](https://html.spec.whatwg.org/multipage/origin.html#relaxing-the-same-origin-restriction).
+The [current discussion with other browser
+vendors](https://github.com/w3ctag/design-reviews/issues/564) is moving toward
+the same direction.
+
+For example, if a hosting service provides different subdomains per user, an
+attacker can set `document.domain` to pretend as if they are the same-origin
+with another user's page. Further, an attacker can host a website under a shared
+hosting service, which serves through the same IP address with different port
+numbers, the attacker can pretend to be on the same-site-but-same-origin as
+yours. This is possible because `document.domain` ignores the port number part
+of the domain.
```

Comment:
```suggestion
For example, if a hosting service provides different subdomains per user, an
attacker can set `document.domain` to pretend they are the same-origin
as another user's page. Further, an attacker can host a website under a shared
hosting service, which serves sites through the same IP address with different port
numbers. In that case, the attacker can pretend to be on the same-site-but-same-origin as
yours. This is possible because `document.domain` ignores the port number part
of the domain.
```